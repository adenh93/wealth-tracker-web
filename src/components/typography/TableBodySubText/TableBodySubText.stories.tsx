import TableBodySubText from './TableBodySubText'
import { Story } from '@storybook/react'

const config = {
  title: 'Components/Typography/TableBodySubText',
  component: TableBodySubText,
}

const Template: Story = (args) => <TableBodySubText {...args} />

export const Primary = Template.bind({})

Primary.args = {
  children: 'This is a Table Body SubText component',
}

export default config
