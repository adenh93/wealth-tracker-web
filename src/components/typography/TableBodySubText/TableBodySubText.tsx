import { Typography } from '@mui/material'
import { FC } from 'react'

const TableBodySubText: FC = ({ children }) => (
  <Typography variant="tableCell2" component="p">
    {children}
  </Typography>
)

export default TableBodySubText
