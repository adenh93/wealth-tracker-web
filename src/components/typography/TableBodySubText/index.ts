export { default } from './TableBodySubText'
export type { TableBodySubTextProps } from './TableBodySubText'