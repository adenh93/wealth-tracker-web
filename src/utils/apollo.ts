import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
