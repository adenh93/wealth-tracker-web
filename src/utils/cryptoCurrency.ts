import { CMC_LOGO_URL } from '../constants'

export const getCMCLogoUrl = (id: number): string => `${CMC_LOGO_URL}/${id}.png`
