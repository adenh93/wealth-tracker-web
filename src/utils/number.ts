const defaultOpts: Intl.NumberFormatOptions = {
  maximumFractionDigits: 2,
  notation: 'standard',
  signDisplay: 'never',
}

export const formatNumber = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  new Intl.NumberFormat('en-US', {
    ...opts,
  }).format(value)

export const formatCurrency = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  formatNumber(value, { ...opts, style: 'currency', currency: 'AUD' })

export const formatPercent = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string => formatNumber(value, { ...opts, style: 'percent' })
