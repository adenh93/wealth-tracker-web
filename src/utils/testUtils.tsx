import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
