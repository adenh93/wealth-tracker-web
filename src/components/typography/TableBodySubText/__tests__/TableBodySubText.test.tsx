import { render } from '../../../../utils/testUtils'
import TableBodySubText from '../TableBodySubText'

describe('TableBodySubText component tests', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<TableBodySubText>Test</TableBodySubText>)

    expect(asFragment).toMatchSnapshot()
  })

  test('renders correctly', () => {
    const { getByText } = render(<TableBodySubText>Test</TableBodySubText>)

    expect(getByText('Test')).not.toBeNull()
  })
})
