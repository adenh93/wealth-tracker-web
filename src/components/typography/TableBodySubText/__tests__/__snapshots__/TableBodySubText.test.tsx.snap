// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodySubText component tests matches snapshot 1`] = `[Function]`;
